src/buffer/message_view.rs

```diff
@@ -0,0 +1,288 @@
+use crate::buffer::scroll_view::Message;
+use data::isupport::CaseMap;
+use data::server::Server;
+use data::target::{self};
+use data::{Config, User, message};
+use iced::advanced::text;
+use iced::widget::{column, container, row};
+
+use super::scroll_view::LayoutMessage;
+use super::user_context;
+use crate::widget::{
+    Element, message_content, message_marker, selectable_text,
+};
+use crate::{Theme, theme};
+
+#[derive(Clone, Copy)]
+pub struct ChannelQueryLayout<'a> {
```

Comment:
Perhaps "`BufferLayout`"?

Reply:
This doesn't layout all buffers though. Log, server and highlight buffers are excluded

Reply:
Good point yeah.